domains/runtime/evm/src/lib.rs

```diff
@@ -175,6 +267,11 @@ impl fp_self_contained::SelfContainedCall for RuntimeCall {
         dispatch_info: &DispatchInfoOf<RuntimeCall>,
         len: usize,
     ) -> Option<TransactionValidity> {
+        if !is_create_contract_allowed(self, &(*info).into()) {
+            // TODO: should this be Custom() instead?
```

Comment:
is the check required here if the extension already does that check ?

Validate should fail to exclude into TXPool.

Intersting point to note here is if operator forcefully adds the txn into bundle, pre_dispatch will fail but operator is not penalized.

I think we should also penilize the operators who allows contract creation calls into bundle by marking them invalid

We currently do that inherent extrinsics, but no reason we cannot do that for this one too